spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/trace/StackdriverTraceAutoConfiguration.java

```diff
@@ -253,11 +231,10 @@ public BaggagePropagation.FactoryBuilder baggagePropagationFactoryBuilder() {
     return BaggagePropagation.newFactoryBuilder(StackdriverTracePropagation.newFactory(primary));
   }
 
-  // Add this bean to suppress other encoding schema, e.g., JSON.
-  @Bean
-  @ConditionalOnMissingBean
-  public BytesEncoder<Span> spanBytesEncoder() {
-    return SpanBytesEncoder.PROTO3;
+  @Bean(ENCODER_BEAN_NAME)
+  @ConditionalOnMissingBean(name = ENCODER_BEAN_NAME)
+  public BytesEncoder<MutableSpan> spanBytesEncoder() {
+    return new StackdriverV2Encoder(Tags.ERROR);
```

Comment:
When running `TraceSampleApplicationIntegrationTests`, the following exception occurs:
```
16:02:55.788 [AsyncReporter{StackdriverSender{projects/[PROJECT]}}] WARN  z.r.internal.AsyncReporter$Flusher - Unexpected error flushing spans
java.lang.NullPointerException: Cannot invoke "String.length()" because "ipString" is null
        at com.google.common.net.InetAddresses.ipStringToBytes(InetAddresses.java:185)
        at com.google.common.net.InetAddresses.forString(InetAddresses.java:151)
        at zipkin2.reporter.stackdriver.brave.AttributesExtractor.extract(AttributesExtractor.java:76)
        at zipkin2.reporter.stackdriver.brave.SpanTranslator.translate(SpanTranslator.java:80)
        at zipkin2.reporter.stackdriver.brave.StackdriverV2Encoder.translate(StackdriverV2Encoder.java:59)
        at zipkin2.reporter.stackdriver.brave.StackdriverV2Encoder.sizeInBytes(StackdriverV2Encoder.java:31)
        at zipkin2.reporter.stackdriver.brave.StackdriverV2Encoder.sizeInBytes(StackdriverV2Encoder.java:15)
        at zipkin2.reporter.internal.CountBoundedQueue.doDrain(CountBoundedQueue.java:115)
        at zipkin2.reporter.internal.CountBoundedQueue.drainTo(CountBoundedQueue.java:83)
        at zipkin2.reporter.internal.AsyncReporter$BoundedAsyncReporter.flush(AsyncReporter.java:241)
        at zipkin2.reporter.internal.AsyncReporter$Flusher.run(AsyncReporter.java:352)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Exception in thread "AsyncReporter{StackdriverSender{projects/[PROJECT]}}" java.lang.NullPointerException: Cannot invoke "String.length()" because "ipString" is null
        at com.google.common.net.InetAddresses.ipStringToBytes(InetAddresses.java:185)
        at com.google.common.net.InetAddresses.forString(InetAddresses.java:151)
        at zipkin2.reporter.stackdriver.brave.AttributesExtractor.extract(AttributesExtractor.java:76)
        at zipkin2.reporter.stackdriver.brave.SpanTranslator.translate(SpanTranslator.java:80)
        at zipkin2.reporter.stackdriver.brave.StackdriverV2Encoder.translate(StackdriverV2Encoder.java:59)
        at zipkin2.reporter.stackdriver.brave.StackdriverV2Encoder.sizeInBytes(StackdriverV2Encoder.java:31)
        at zipkin2.reporter.stackdriver.brave.StackdriverV2Encoder.sizeInBytes(StackdriverV2Encoder.java:15)
        at zipkin2.reporter.internal.CountBoundedQueue.doDrain(CountBoundedQueue.java:115)
        at zipkin2.reporter.internal.CountBoundedQueue.drainTo(CountBoundedQueue.java:83)
        at zipkin2.reporter.internal.AsyncReporter$BoundedAsyncReporter.flush(AsyncReporter.java:241)
        at zipkin2.reporter.internal.AsyncReporter$Flusher.run(AsyncReporter.java:352)
        at java.base/java.lang.Thread.run(Thread.java:1583)
```

The [AttributesExtractor logic](https://github.com/openzipkin/zipkin-gcp/blob/481cc1cde2eb2962d0d91dd183e30831a43db25f/encoder-stackdriver-brave/src/main/java/zipkin2/reporter/stackdriver/brave/AttributesExtractor.java#L76):
```java
    if (braveSpan.localServiceName() != null && braveSpan.kind() == Span.Kind.SERVER) {
      // Create an IP without querying DNS
      InetAddress ip = InetAddresses.forString(braveSpan.localIp()); // Note: localIp() is null at this point.
```

`zipkin2.reporter.internal.report(next)` pushes the span instance to the pending queue w/o a `localIp`... but I am currently unable to find who in the creation logic chain is responsible for setting the `localIp` value in the ideal case...

Perhaps of note... traveling up the stacktrace @ `org.springframework.web.filter.ServerHttpObservationFilter`, `observation.context.response.response.request.remoteAddr = 127.0.0.1`. It then travels through `io.micrometer.observation`, then `io.micrometer.tracing`, then `brave`, then `zipkin2.reporter`.